lxd/api_cluster.go

```diff
@@ -1568,9 +1583,39 @@ func updateClusterNode(d *Daemon, r *http.Request, isPatch bool) response.Respon
 		return response.SmartError(err)
 	}
 
-	// If cluster roles changed, then distribute the info to all members.
+	// Check for changes in node roles.
 	if state.Endpoints != nil && clusterRolesChanged(node.Roles, newRoles) {
+		// If cluster roles changed, then distribute the info to all members.
 		cluster.NotifyHeartbeat(state, d.gateway)
+
+		// Check if OVN chassis role was added/removed.
+		hasRole := false
+		for _, role := range newRoles {
+			if role == db.ClusterRoleOVNChassis {
+				hasRole = true
+				break
+			}
+		}
+
+		hadRole := false
+		for _, role := range node.Roles {
+			if role == db.ClusterRoleOVNChassis {
+				hadRole = true
+				break
+			}
+		}
+
+		if (hadRole && !hasRole) || (!hadRole && hasRole) {
+			client, err := cluster.Connect(nodeInfo.Address, d.endpoints.NetworkCert(), d.serverCert(), r, true)
```

Comment:
This notification sections feels redundant now that we are doing it via the heartbeat notification above.
What is possibly missing is the restarting of OVN networks on this local member.

Handling this inside `NotifyHeartbeat()` would be more consistent with how the cluster event listeners are refreshed in https://github.com/stgraber/lxd/blob/66a4a76dae719c65051671ccb79dcec1a5569c76/lxd/cluster/membership.go#L621-L626

Doing it in there also lends credance to my idea above of moving away from the `d.lastNodeList` variable in preference of per-task refresh functions. If we had an `networkUpdateOVNChassisTask(d.State(), heartbeatData)` function that maintained via its own last state var how it was operating, then it could be called both from within `cluster.NotifyHeartbeat` and from within `nodeRefreshTask`.